gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobReadChannel.java

```diff
@@ -46,4 +45,27 @@
    */
   void chunkSize(int chunkSize);
 
+  /**
+   * Saves the read channel state.
+   *
+   * @return an object that contains the read channel state and can restore it afterwards. State
+   *     object must implement {@link java.io.Serializable}.
+   */
+  public State save();
+
+  /**
+   * A common interface for all classes that implement the internal state of a
+   * {@code BlobReadChannel}.
+   *
+   * Implementations of this class must implement {@link java.io.Serializable} to ensure that the
+   * state of a channel can be correctly serialized.
+   */
+  public interface State {
```

Comment:
Do you think this could have been defined as a top level (maybe even in core) and used for both using generics

```java
public interface State<T> {
  T restroe();
}
```